src/EpiContentUsage/Frontend/Views/ContentTypeUsageView.tsx

```diff
@@ -1,441 +1,448 @@
-import React, { useCallback, useEffect, useMemo, useState } from "react";
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
 import { Breadcrumb } from "@episerver/ui-framework";
-import { SortDirection } from "../types";
+import { TableColumn } from "../types";
 import {
-  Button,
+  ButtonIcon,
+  DiscloseTable,
   Dropdown,
   Grid,
   GridCell,
   GridContainer,
-  Input,
   Link,
   PaginationControls,
   Table,
-  Typography,
 } from "optimizely-oui";
 import { useLoaderData, useLocation } from "react-router-dom";
-import { AxiosResponse } from "axios";
 import Layout from "../Components/Layout";
-import { getRoutePath, viewContentTypes } from "../routes";
+import { getRoutePath, navigateTo, viewContentTypes } from "../routes";
 import { ContentTypeDto, ContentUsageDto } from "../dtos";
+import { useFilteredTableData } from "../Lib/hooks/useFilteredTableData";
+import { APIResponse } from "../Lib/EpiContentUsageAPIClient";
+import Header from "../Components/Header";
+import { useTranslations } from "../Contexts/TranslationsProvider";
+import Filters from "../Components/Filters/Filters";
 
-type TableColumn = "guid" | "id" | "name" | "languageBranch" | "pageUrl";
+enum ContentTypeUsageTableColumn {
+  ID = "id",
+  Name = "name",
+  LanguageBranch = "languageBranch",
+  PageUrl = "pageUrl",
+  Actions = "actions",
+}
 
-const ROWS_PER_PAGE_OPTIONS = [15, 30, 60];
+interface ContentTypeUsageTableRowProps<TableDataType> extends ContentUsageDto {
+  tableColumns: TableColumn<TableDataType>[];
+  onRowClick?: (event: React.PointerEvent) => void;
+  onEditButtonClick?: (event: React.PointerEvent) => void;
+  onViewWebsiteClick?: () => void;
+  hasDiscloseTableRows?: boolean;
+}
+
+const ContentTypeUsageDiscloseTableRow = ({
```

Comment:
Disclose row doesn't display action buttons. I propose to create new issue for this.